### DISCUSSION TITLE
Support Forgejo as a standalone option (instead of via Gitea)

### DISCUSSION THREAD

**Discussion by filipdutescu** [fluxcd/flux2#4924] [Proposals] [answered]
As a user of [Forgejo][forgejo], I want to be able to use a repository from a forge to be used by Flux. [Forgejo][forgejo] is a (now) hard-fork of [Gitea][gitea] which aims to preserve compatibility with it, as much as possible.

Hopefully, this means it should not be too hard to add support for [Forgejo][forgejo] and that, at least for the foreseeable future, it should not be too much of a burden, as [Gitea][gitea] is already supported.

[forgejo]: https://forgejo.org/
[gitea]: https://about.gitea.com/

**Comment by ancashoria** [comment]
Did support for Forgejo land in Flux?

**Reply by matheuscscp** [reply, marked as answer]
Flux works with all Git providers. Support is based on standard protocols: HTTPS and SSH.

This issue is not specific about any particular Git integrations in Flux (e.g. source-controller and image-automation-controller, or notification-controller), so here is the answer for each one:

* For source-controller and image-automation-controller, they work with all Git providers. Support is based on standard protocols: Git over HTTPS and Git over SSH.
* For notification-controller, which integrates with Git provider APIs (not Git protocols), we do support Gitea, and I believe this immediately gives you support for Forgejo. Please test and let us know so we can close this discussion.
